### PR TITLE
Resolve clean build version

### PR DESCRIPTION
**Pull request by CamaradeRoman**
Example for monit-unit component:
* versionPrefix = monit-unit
* lineVersionFormat = majorVersionFormat = '$major'
* releaseVersionFormat = buildVersionFormat = '$major.$minor'

jiraComponentVersion.version resolving by component (in rest/api/2/components/${component}/versions/${version}/jira-component for instance):

| Version parameter | Old resolved version | New resolved version |
|-------------------|----------------------|----------------------|
| 1                 | 1                    | 1                    |
| 1.2               | 1.2                  | 1.2                  |
| 1-2               | 1-2                  | 1.2                  |
| 1.2.3             | 1.2.3                | 1.2                  |
| 1.2.3.4           | 1.2.3.4              | 1.2                  |
| 1.02              | 1.02                 | 1.2                  |
| 1_RC              | 1_RC                 | 1                    |
| 1.2_RC            | 1.2_RC               | 1.2                  |
| 1-SNAPSHOT        | 1-SNAPSHOT           | 1                    |
| 1.2-SNAPSHOT      | 1.2-SNAPSHOT         | 1.2                  |
| test-1            | test-1               | 1                    |
| test-1.2          | test-1.2             | 1.2                  |
| monit-unit-1      | 1                    | 1                    |
| monit-unit-1.2    | 1.2                  | 1.2                  |
| monit-unit-1_RC   | monit-unit-1_RC      | 1                    |
| monit-unit-1.2_RC | monit-unit-1.2_RC    | 1.2                  |

jiraComponentVersion.version resolving by project (in rest/api/2/projects/${project}/versions/${version} for instance):

| Version parameter         | Old resolved version | New resolved version |
|---------------------------|----------------------|----------------------|
| 1                         | Error                | Error                |
| 1.2                       | Error                | Error                |
| 1.2.3                     | Error                | Error                |
| monit-unit-1              | 1                    | 1                    |
| monit-unit-1.2            | 1.2                  | 1.2                  |
| monit-unit-1.2.3          | Error                | Error                |
| monit-unit-1_RC           | Error                | Error                |
| monit-unit-1.2_RC         | monit-unit-1.2_RC    | 1.2                  |
| monit-unit-1.2.3_RC       | Error                | Error                |
| monit-unit-1-SNAPSHOT     | Error                | Error                |
| monit-unit-1.2-SNAPSHOT   | Error                | Error                |
| monit-unit-1.2.3-SNAPSHOT | Error                | Error                |

### CODE REVIEW

**Review by qu0d — DISMISSED** [review]

File: components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/impl/ComponentRegistryResolverImpl.kt [hunk 1]
```diff
@@ -69,127 +70,74 @@ class ComponentRegistryResolverImpl(
...
+    override fun getJiraComponentVersionRangesByProject(projectKey: String) =
+        jiraParametersResolver.componentConfig.projectKeyToJiraComponentVersionRangeMap[projectKey]?.toSet()
+            ?: throw NotFoundException("Project '$projectKey' is not found")
 
     override fun getComponentsDistributionByJiraProject(projectKey: String): Map<String, Distribution> {
-        return getProjectJiraComponentVersionRanges(projectKey)
+        return getJiraComponentVersionRangesByProject(projectKey)
             .groupBy { it.componentName }
-            .mapValues {
+            .mapValues { entry ->
```
Comment (aryabokon) [inline components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/impl/ComponentRegistryResolverImpl.kt:111]: may be use destruction like (_, value) -> ?

Comment (CamaradeRoman) [reply components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/impl/ComponentRegistryResolverImpl.kt]: I've just fixed hiding of `it` here. But destruction is better indeed. Fixed.

**Review by pgorbachev — CHANGES_REQUESTED** [review]
It might worth adding test cases ComponentRegistryResolver, especially for the the part that is affected by refactoring

**Comment by CamaradeRoman** [comment]
> It might worth adding test cases ComponentRegistryResolver, especially for the the part that is affected by refactoring

Done

**Review by pgorbachev — APPROVED** [review]

**Review by aryabokon — APPROVED** [review]

**Pull request merged**